WordPress/gutenberg
# [Mobile] Create endpoint to post notices from host apps

**Fixes:** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2453

**Related PRs:** 
`gutenberg-mobile` https://github.com/wordpress-mobile/gutenberg-mobile/pull/2613
`WordPress-iOS` https://github.com/wordpress-mobile/WordPress-iOS/pull/14886

## Description
Adds an api that the host apps can use to present the info notice. This is being used as part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/2453 to show the page created messages. 
🗒️ The linked PRs cover the iOS version of this feature. @antonis will be working on the Android side in a later PR.

## How has this been tested?
🗒️ Test cases here are taken from https://github.com/wordpress-mobile/WordPress-iOS/pull/14886
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

Start by navigating to the Modal Layout Picker

### Page Created (with content)
1. Select a Layout from the options in Modal Layout Picker
1. Select "Create Page"
**Expect** to see the info notice banner appear under the navigation bar with the text "Page created"
- 🗒️ This is different than the mocks in the linked issue but aligns with the direction that was determined. 

### Blank Page Created
1. Select "Create Blank Page"
**Expect** to see the info notice banner appear under the navigation bar with the text "Blank page created"
- 🗒️ This is different than the mocks in the linked issue but aligns with the direction that was determined. 

### Blog Post Created
1. Create a new blog post
- This should not display the notice bar

### Open existing Page Created
1. Open an existing page (either published or draft)
- This should not display the notice bar

### Disable Modal Layout Picker
1. Create a new page
- This should not display the notice bar

## Screenshots 

| Create Page | Create Blank Page | Clip |
| --- | --- | --- |
| <img width = 300 src="https://user-images.githubusercontent.com/3384451/92971358-4f1abc00-f44e-11ea-92f7-49d54ac8bd0c.png" /> | <img width = 300 src="https://user-images.githubusercontent.com/3384451/92971288-30b4c080-f44e-11ea-994e-9c43c9b297b0.png"/> |  <img width = 300 src="https://user-images.githubusercontent.com/3384451/92971278-2d213980-f44e-11ea-8736-e0f433e8247c.gif"/> |

## Types of changes




Enhancement

## Checklist:
